Update MangaWorld presence

TheTecnoKing opened the pull request:
Remade the MangaWorld presence to support their new website.
### Screenshots:
<details>
<summary>Homepage</summary>

![homepage](https://i.imgur.com/7Bik4Yc.png)
</details>

<details>
<summary>Bookmarks</summary>

*   Main ![bookmarks](https://i.imgur.com/6zVbMQ5.png)
*   Reading ![bookmarks_reading](https://i.imgur.com/Sy64xAO.png)
*   Completed ![bookmarks_completed](https://i.imgur.com/SdFGDy6.png)
*   Paused ![bookmarks_paused](https://i.imgur.com/nd8MQKy.png)
*   Plan to read ![bookmarks_plantoread](https://i.imgur.com/Bi8P5bW.png)
*   Dropped ![bookmarks_dropped](https://i.imgur.com/FHdKZSY.png)
*   Re-reading ![bookmarks_rereading](https://i.imgur.com/s2puB8u.png)
</details>

<details>
<summary>Archive</summary>

*   Main ![archive](https://i.imgur.com/Zm1rYmw.png)
*   Search ![archive_search](https://i.imgur.com/5fpecRV.png)
*   Sort by artist ![archive_artist](https://i.imgur.com/1AzVCpw.png)
*   Sort by author ![archive_author](https://i.imgur.com/WvkYE7M.png)
*   Sort by genre ![archive_genre](https://i.imgur.com/vNsxN7e.png)
*   Sort by type ![archive_type](https://i.imgur.com/6M9yZ8m.png)
*   Sort by status ![archive_status](https://i.imgur.com/ypTyAXh.png)
*   Sort by year ![archive_year](https://i.imgur.com/1Ol82RH.png)
*   Order by ![archive_orderby](https://i.imgur.com/kTlrt5h.png)
</details>

<details>
<summary>Reading</summary>

*   Main ![reading](https://i.imgur.com/NCwfNgh.png)
*   In the e-reader ![reading_ereader](https://i.imgur.com/UtSyJi7.gif) *Broken embed due to "Content length exceeded", please click [this](https://i.imgur.com/UtSyJi7.gif) instead.*
*   "Read along!" button ![reading_readalong](https://i.imgur.com/YcylRXx.gif)
</details>

<details>
<summary>Other pages</summary>

![otherpages](https://i.imgur.com/OUtVZWo.png)
</details>

TheTecnoKing committed:
Modify and update presence to support the new website

- Updated all the assets
- Updated the base URL
- Updated the old, partially working presences (bookmarks, homepage, parts of reading, parts of the archive)
- Removed code referring to website sections that don't exist anymore
- Added comments for all the presences (written in Italian and English)
- Added new presences for archive, reading, bookmarks

TheTecnoKing committed:
Merge branch 'PreMiD:master' into master

TheTecnoKing committed:
Merge branch 'PreMiD:master' into master

TheTecnoKing committed:
Remove setActivity for every check

- Removed setActivity in previous checks and added one at the end
- Edited the homepage's value to check: from "https://www.mangaworld.io" to pathname == "/". This was done in order to keep the presence working by only editing the metadata in case the site changes domain.

TheTecnoKing committed:
Modify version number

Forgot to modify the version number in the last commit... Oops

TheTecnoKing committed:
Update presence's strings

Updated the presence's strings:
- Modified strings for section: "Bookmarks"
- Modified strings for section: "Order by"
- Modified strings for section: "Other pages"

TheTecnoKing committed:
Update a string (again)

TheTecnoKing committed:
Separate undefined pages' presence from manga pages' presence

- Created a new check for pages that haven't been defined earlier in the code. Those pages used to show up in the manga pages' presence, thus making it hard to write a common string.
- Cleaned the code a bit

TheTecnoKing committed:
Remove too many setTimestamp

Removed setTimestamps in checks and added one at the end of the file

TheTecnoKing committed:
Fix strings

code-factor committed:
[CodeFactor] Apply fixes

Bas950 commented on websites/M/MangaWorld/dist/metadata.json:
```diff
@@ -3,24 +3,30 @@
-  "logo": "https://i.imgur.com/MdYzAHr.png",
-  "thumbnail": "https://i.imgur.com/BFsORve.png",
-  "color": "#EE6B00",
+  "version": "3.0.6",
```
```suggestion
  "version": "3.0.0",
```

Bas950 reviewed (changes requested):
Fix errors reported by the bots

TheTecnoKing committed:
Fix most issues reported by CodeFactor

Some issues have been left unfixed, checking those later.

TheTecnoKing committed:
Update websites/M/MangaWorld/dist/metadata.json

Co-authored-by: Bas950 <bas.van.zanten007@gmail.com>

TheTecnoKing commented:
Do I have to fix all the indentation and curly brace warnings as well?
I left the braces not on the same line with the next block on purpose, so I could write comments for each section.

TheTecnoKing committed:
Fix issues requiring to use array destructuring

Bas950 commented:
> Do I have to fix all the indentation and curly brace warnings as well?
> I left the braces not on the same line with the next block on purpose, so I could write comments for each section.
> @Bas950

Yes, please use our prettier settings etc.
You can also put the comment in the first line after the if case

TheTecnoKing committed:
Prettify presence.ts

- Prettified presence.ts and fixed indentation issues reported by ESLint
- Updated two strings

TheTecnoKing committed:
Merge branch 'PreMiD:master' into master

TheTecnoKing committed:
Merge branch 'PreMiD:master' into master

TheTecnoKing committed:
Add "Go to manga's main page" button

Added a button that redirects to the manga's dedicated page.
> Manga-dedicated pages show common details about them: plot, genres, author, artist, etc...

TheTecnoKing committed:
Merge branch 'master' of https://github.com/TheTecnoKing/Presences

Bas950 commented on websites/M/MangaWorld/presence.ts:
```diff
@@ -1,193 +1,201 @@
+    data.details = "Visualizzando la pagina:";
+    data.state = `"${pageName}"`;
+  }
+  if (data.details !== null) {
```
```suggestion
  if (data.details) {
```

TheTecnoKing committed:
Update websites/M/MangaWorld/presence.ts

Remove "!== null" when checking for data.details

Co-authored-by: Bas950 <bas.van.zanten007@gmail.com>

TheTecnoKing committed:
Merge branch 'PreMiD:master' into master

TheTecnoKing committed:
Merge branch 'PreMiD:master' into master
